### PR TITLE
prevent using non-object component 

### LINKED ISSUE #3068: JavaScript error when submitting form when cf2 component not set.
Matt reports : `uncaught type error h.getAllFieldValues()`

https://imgur.com/sdxTIVG

I bet this is caused here: https://github.com/CalderaWP/Caldera-Forms/blob/master/clients/render/index.js#L71

#3049 ?

### PR DESCRIPTION
**Pull request by Shelob9**
Fixes #3068

Needs tests

### CODE REVIEW

**Comment by Shelob9** [comment]
a14b094 was committed to wrong branch, I am reverting.

**Comment by New0** [comment]
@Shelob9 we're not in a testable function. We need that condition here as the js console error seems to come from index.js l79 ( theComponent.getAllFieldValues(); )
We could run the same condition again in processFiles function in fileUploads.js, this would be testable ( but not real...)

File: clients/render/index.js [hunk 1]
```diff
@@ -62,8 +62,9 @@ domReady(function () {
 
 
 		jQuery(document).on('cf.ajax.request', (event, obj) => {
+
 			//do not run if component is empty
-			if( ! theComponent ){
+			if( theComponent.props.fieldsToControl.length <= 0 ){
```
Comment (Shelob9) [inline clients/render/index.js]: @New0 When I tested, the problem was `theComponent` was null. 

If you want to test `theComponent.props.fieldsToControl.length` fine, but first test that props and props.fieldsControl are properties and theComponent is an object.

**Comment by New0** [comment]
@Shelob9 thank you for your review

Also, we're not in a testable function, I believe we need his condition here because of theComponent.getAllFieldValues(); l81
We could re-run the condition inside processFileFields in fileUploads.js, that would be testable, but not real

File: clients/render/index.js [hunk 2]
```diff
@@ -63,11 +63,15 @@ domReady(function () {
 
 		jQuery(document).on('cf.ajax.request', (event, obj) => {
 
-			//do not run if component is empty
-			if( theComponent.props.fieldsToControl.length <= 0 ){
+			//do not run if component don't have fieldsToControl
+			if( ! theComponent ){
 				return;
+			} else if( typeof theComponent === "object" ){
+				if(theComponent.props.fieldsToControl.length <= 0){
```
Comment (Shelob9) [inline clients/render/index.js]: @New0 If `theComponent` is a `null` then your conditional on 69 will pass. Seams silly but try it https://jsfiddle.net/ouejgzrm/

Why not `'object' === type of theComponent && 'function' === typeof theComponent.getAllFieldValues()` ?

**Comment by New0** [comment]
@Shelob9 Yes, I wasn't thinking this through, comparing against the function itself makes sense.
Let me know if the new condition is still not safe enough

**Comment by swapnilmanjrekar** [comment]
@Shelob9 @New0 
- **According to my testing & findings, I see this PR doesn't solves the Issue as I encountered JavaScript errors when loading and submitting my contact form and form with one advanced file field and also I was not able to see the advance file field v2.0 field in my Form with one advanced file field v2.0. field**


**I checked this PR using the following steps:**

- I created a  WordPress test site on my desktop locally using Lando.
- I installed Caldera Forms plugin on my test site.
- I switched to the feature/3068 branch of Caldera Forms.
- I created a contact us form from the available form templates.
- I saved the created contact us form.
- Next, I created a form with one advanced file field v2.0 field and submit button.
- I saved the created form.
- Next, I Started testing for Test Scenarios for contact us form and for a form with one advanced file field version 2.0
- I tested this PR on Google Chrome,Mozilla Firefox and Microsoft Edge Browsers.

1. **For Contact Us Form:**

- I checked this on my local Lando WordPress test site.
- I tested this on feature/3068 branch of Caldera Forms.
- I loaded the created Contact Us Form.
- I entered all the details in my Contact Us Form.
- I expected to see that after entering all details in my Contact Us form & when clicking on submit button, form getting submitted.
- But when I entered all the details in my contact form and clicked on submit button of my contact form, the form was not getting submitted and it was not as per my expectations.

2. **For a Form with one advanced file field version 2.0:**

- I checked this on my local Lando WordPress test site.
- I tested this on feature/3068 branch of Caldera Forms.
- I loaded the created Form with one advanced file field.
- I expected to see one advanced file field version 2.0 along with submit button in my form
- After loading my form with one advanced file field version 2.0, I was not able to see the advanced file field v2.0 field in my form with one advanced file field version 2.0 field and it was not as per my expectations.

**PHP Errors:**
- For Contact Form: When I checked the debug.log file of  my local WordPress test site I didn't see any PHP errors. 
- For Form with one advanced file field version 2.0: When I checked the debug.log file of my local WordPress test site I didn't see any PHP errors

**JavaScript Errors:**
- For Contact Form: When I checked the console of my browser I saw the following JavaScript error:
  " Uncaught TypeError: Cannot read property 'Component' of undefined
    at Object.709 (index.js:507)
    at __webpack_require__ (bootstrap 85bd2c149c82c4442d1c:19)
    at Object.706 (index.min.js?ver=1.8.2:4398)
    at __webpack_require__ (bootstrap 85bd2c149c82c4442d1c:19)
    at Object.247 (CalderaFormsFieldGroup.js:1)
    at __webpack_require__ (bootstrap 85bd2c149c82c4442d1c:19)
    at Object.185 (CalderaFormsFieldRender.js:1)
    at __webpack_require__ (bootstrap 85bd2c149c82c4442d1c:19)
    at Object.688 (index.min.js?ver=1.8.2:3294)
    at __webpack_require__ (bootstrap 85bd2c149c82c4442d1c:19)
    TypeError: Cannot read property 'CF5c5b12c48f77e_1' of undefined
    at Object.init (caldera-forms-front.min.js?ver=1.8.2:1)
    at HTMLFormElement.<anonymous> (jquery-baldrick.min.js?ver=1.8.2:1)
    at HTMLFormElement.dispatch (jquery.js?ver=1.12.4:3)
    at HTMLFormElement.r.handle (jquery.js?ver=1.12.4:3)"

- For Form with one advanced file field version 2.0: When I checked the console of my browser, I saw the following JavaScript error:
  " Uncaught TypeError: Cannot read property 'Component' of undefined
    at Object.709 (index.js:507)
    at __webpack_require__ (bootstrap 85bd2c149c82c4442d1c:19)
    at Object.706 (index.min.js?ver=1.8.2:4398)
    at __webpack_require__ (bootstrap 85bd2c149c82c4442d1c:19)
    at Object.247 (CalderaFormsFieldGroup.js:1)
    at __webpack_require__ (bootstrap 85bd2c149c82c4442d1c:19)
    at Object.185 (CalderaFormsFieldRender.js:1)
    at __webpack_require__ (bootstrap 85bd2c149c82c4442d1c:19)
    at Object.688 (index.min.js?ver=1.8.2:3294)
    at __webpack_require__ (bootstrap 85bd2c149c82c4442d1c:19)"


WordPress Version: 5.1 PHP Version: 7.1.26 MySQL Version: 5.7.25 Caldera Forms Version: 1.8.2 WP_DEBUG: 1

**Pull request merged**